packages/embark-specialconfigs/src/index.js

```diff
@@ -18,6 +18,7 @@ class SpecialConfigs {
     this.embark.registerActionForEvent('deployment:deployContracts:beforeAll', this.beforeAllDeployAction.bind(this));
     this.embark.registerActionForEvent('deployment:deployContracts:afterAll', this.afterAllDeployAction.bind(this));
     this.embark.registerActionForEvent("deployment:contract:deployed", this.doOnDeployAction.bind(this));
+    this.embark.registerActionForEvent("deployment:contract:undeployed", this.doOnDeployAction.bind(this));
```

Comment:
Not sure if the onDeploy should run on `deployment:contract:undeployed`. Since.. well it's `onDeploy`.

My rationale is that usually, in `onDeploy` you would call something like setting some initial value or something like that so you probably don't want to call that every time.

At the very least, it would keep the same functionality as before (we never executed `onDeploy` when undeployed.

Reply:
I agree, my goal here was to maintain feature parity with 4.x, but I could have been mistaken that `onDeploy` was executed for non-deployed contracts. Removed.